syft/pkg/cataloger/apkdb/parse_apk_db.go

```diff
@@ -161,3 +181,42 @@ func parseApkDBEntry(reader io.Reader) (*pkg.ApkMetadata, error) {
 
 	return &entry, nil
 }
+
+func discoverPackageDependencies(pkgs []*pkg.Package) (relationships []artifact.Relationship) {
+	// map["provides" string] -> packages that provide the "p" key
+	lookup := make(map[string][]*pkg.Package)
+	// read "Provides" (p) and add as keys for lookup keys
+	for _, p := range pkgs {
+		apkg, ok := p.Metadata.(pkg.ApkMetadata)
+		if !ok {
+			// TODO warn
+			continue
+		}
+		for _, provides := range apkg.Provides {
+			// TODO: parse provides to remove =*
```

Comment:
It looks like there are still some TODOs here

Reply:
cc @wagoodman for context from our gardening meeting

Reply:
while building this lookup, each string isn't necessarily a package reference. It could be a package name at a version: https://github.com/anchore/syft/blob/ed5c9c9b250615608da27a553e2ba404311b3d7c/syft/pkg/cataloger/apkdb/test-fixtures/extra-file-attributes#L14
```
so:openjdk8:libfontmanager.so=0
```
we need to extract out the names and ignore the versions:
```
so:openjdk8:libfontmanager.so
```